File: app/src/main/java/jp/superwooo/chaipon/lighttimeswitcher/EnableTimePreference.java
package jp.superwooo.chaipon.lighttimeswitcher;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.preference.PreferenceManager;

import java.time.LocalTime;

public class EnableTimePreference {
    private Context mContext;
    private final String mHourKey;
    private final String mMinuteKey;
    private final String mEnabledKey;
    private SharedPreferences mPreference;
    private String mKeyPrefix;

    public static EnableTimePreference Create(Context context, String keyPrefix){
        return new EnableTimePreference(context, keyPrefix);
    }

    private EnableTimePreference(Context context, String keyPrefix){
        mContext = context;
        mPreference = PreferenceManager.getDefaultSharedPreferences(mContext.getApplicationContext());
        mKeyPrefix = keyPrefix;
        mHourKey = keyPrefix + "hour";
        mMinuteKey = keyPrefix + "minute";
        mEnabledKey = keyPrefix + "enabled";

    }
    public void save(LocalTime time, boolean enabled){
        SharedPreferences.Editor e = mPreference.edit();
        e.putBoolean(mEnabledKey, enabled);
        e.putInt(mHourKey, time.getHour());
        e.putInt(mMinuteKey, time.getMinute());
        e.apply();
    }
    public void save(boolean enabled) {
        SharedPreferences.Editor e = mPreference.edit();
        e.putBoolean(mEnabledKey, enabled);
        e.apply();
    }

    public LocalTime loadTime(){
        int hour = mPreference.getInt(mHourKey, 0);
        int minute = mPreference.getInt(mMinuteKey, 0);
        return LocalTime.of(hour, minute);
    }

    public boolean isEnabled(){
        return mPreference.getBoolean(mEnabledKey, false);
    }
}

File: app/src/main/java/jp/superwooo/chaipon/lighttimeswitcher/EnableTimePreference.kt
package jp.superwooo.chaipon.lighttimeswitcher

import android.content.Context
import android.content.SharedPreferences
import androidx.preference.PreferenceManager
import java.time.LocalTime

class EnableTimePreference private constructor(
    private val context: Context,
    private val keyPrefix: String
) {
    private val hourKey = keyPrefix + "hour"
    private val minuteKey = keyPrefix + "minute"
    private val enabledKey = keyPrefix + "enabled"
    private val preference: SharedPreferences =
        PreferenceManager.getDefaultSharedPreferences(context.applicationContext)

    fun save(time: LocalTime, enabled: Boolean) {
        val e = preference.edit()
        e.putBoolean(enabledKey, enabled)
        e.putInt(hourKey, time.hour)
        e.putInt(minuteKey, time.minute)
        e.apply()
    }

    fun save(enabled: Boolean) {
        val e = preference.edit()
        e.putBoolean(enabledKey, enabled)
        e.apply()
    }

    fun loadTime(): LocalTime {
        val hour = preference.getInt(hourKey, 0)
        val minute = preference.getInt(minuteKey, 0)
        return LocalTime.of(hour, minute)
    }

    val isEnabled: Boolean
        get() = preference.getBoolean(enabledKey, false)

    companion object {
        fun create(context: Context, keyPrefix: String): EnableTimePreference {
            return EnableTimePreference(context, keyPrefix)
        }
    }
}
